### ISSUE TITLE
Sonos: impact of the deprecated thingRegistry in the BaseThingHandler

### ISSUE THREAD

**Issue by lolodomo** [eclipse/smarthome#6007]
I just discovered that the thingRegistry has been deprecated inside the BaseThingHandler.

Due to the group management done by Sonos, the handler of a Sonos thing sometimes needs to get the handler of another Sonos thing (in particular to get the group controller). To do that, the the binding is searching the thing using the thingRegistry.

What would be the solution without the thingRegistry to get the handler of another thing ?

**Comment by lolodomo** [comment]
A bridge thing has a link to its child things. So the solution could be to introduce a bridge thing. But it will introduce mainly confusion and unnecessary complexity for the user.

**Comment by cweitkamp** [comment]
That is a very good question. And the solution will be the final step to solve #5182.

I am not sure if a thing should be allowed to query for other things and access them (expect relation between thing and bridge). If they are I suggest to move the logic into the `ThingHandlerCallback`.

**Comment by kaikreuzer** [comment]
> I am not sure if a thing should be allowed to query for other things

No, they definitely shouldn't.
Note that this question also came up [in the Bose Soundtouch binding](https://github.com/eclipse/smarthome/pull/5397#issuecomment-383119107), where this functionality was removed for this reason. As stated there, I think the correct way to do zone handling (and thus have functionality that goes across different things) would be rule actions.

**Comment by lolodomo** [comment]
Rule actions are out of the scope of a binding ?
Without a minimum group handling, the Sonos binding can just not work.

**Comment by lolodomo** [comment]
The only solution I can imagine is the bridge thing. We could consider that bridge thing as the Sonos network or Sonos solution.

**Comment by kaikreuzer** [comment]
> Rule actions are out of the scope of a binding ?

No, what I was saying is that rule actions are the way to go for such features.

**Comment by lolodomo** [comment]
We are not talking about group management but simple controls like play pause that require a redirect to the group coordinator.

**Comment by kaikreuzer** [comment]
Ah, ok, sorry, I indeed thought about group management.
I agree that somehow a "bridge" solution seems easiest. Could we maybe make any Sonos device that is capable of being a coordinator a bridge and dynamically assign the things to their coordinator? But this leaves the question, which code should do these assignments...

**Comment by lolodomo** [comment]
I think the bridge should just have the knowledge of the groups and should be able to provide a reference to a thing handler requested by a thing.
For example a Sonos thing handler could request the bridge for the handler of the group coordinator.

**Comment by lolodomo** [comment]
That's way there would be a minimim of changes to apply to the code.

**Comment by kgoderis** [comment]
Beware : Any player can become a group coordinator as soon as a Sonos group is formed. In fact, - I think - if not in a group the player is its own coordinator ( group of one)

**Comment by lolodomo** [comment]
Exactly
